### PR TITLE
Generic collision detection test suite

### PR DESCRIPTION
**Pull request by j-petit**
### Description
To share the same test cases across different collision checkers, I changed the originally for FCL designed test cases to a generic typed test suite. Consequently, I also moved the test implementation out of the FCL folder and into the general `collision_detection`. For testing a specific collision checker, one can now simply include the tests and instantiate them with the specific type. 

To make the changes easy to see, take a look at the first commit which contains the changes to the test whereas the second one deals with moving the files.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)

### CODE REVIEW

File: moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+#include <urdf_parser/urdf_parser.h>
+#include <geometric_shapes/shape_operations.h>
+
+#include <gtest/gtest.h>
+#include <sstream>
+#include <algorithm>
+#include <ctype.h>
+#include <fstream>
+
+template <class T>
```
Comment (BryceStevenWilley) [inline moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h]: Make `T` more descriptive here, i.e., `CollisionCheckerType`.

File: moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+    collision_detection::CollisionDetectorAllocatorPtr alloc(new T);
+    crobot_ = alloc->allocateRobot(robot_model_);
+    cworld_ = alloc->allocateWorld(collision_detection::WorldPtr(new collision_detection::World()));
+  }
+
+  void TearDown() override
+  {
+  }
+
+protected:
```
Comment (BryceStevenWilley) [inline moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h]: Remove dup `protected`.

File: moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("pr2");
+    robot_model_ok_ = static_cast<bool>(robot_model_);
+    kinect_dae_resource_ = "package://moveit_resources/pr2_description/urdf/meshes/sensors/kinect_v0/kinect.dae";
+
+    acm_.reset(new collision_detection::AllowedCollisionMatrix(robot_model_->getLinkModelNames(), true));
+
+    collision_detection::CollisionDetectorAllocatorPtr alloc(new T);
```
Comment (BryceStevenWilley) [inline moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h]: Do you need a `new T` here, or can you reference the pointer to `value_`, i.e. `alloc(&value_);`?

Comment (j-petit) [reply moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h]: You're right

File: moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h (hunk 1 above)
Comment (j-petit) [reply moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h]: I changed it to `CollisionAllocatorType`

**Pull request merged**